posthog/tasks/test/test_calculate_cohort.py

```diff
@@ -91,4 +96,75 @@ def test_exponential_backoff(self, patch_increment_version_and_enqueue_calculate
             enqueue_cohorts_to_calculate(5)
             self.assertEqual(patch_increment_version_and_enqueue_calculate_cohort.call_count, 2)
 
+        @patch("posthog.tasks.calculate_cohort.chain")
+        @patch("posthog.tasks.calculate_cohort.calculate_cohort_ch.si")
+        def test_increment_version_and_enqueue_calculate_cohort_with_nested_cohorts(
+            self, mock_calculate_cohort_ch_si: MagicMock, mock_chain: MagicMock
+        ) -> None:
+            # Create leaf cohort A
+            cohort_a = Cohort.objects.create(
+                team=self.team,
+                name="Cohort A",
+                groups=[{"properties": [{"key": "$some_prop", "value": "something", "type": "person"}]}],
+                is_static=False,
+            )
+
+            # Create cohort B that depends on cohort A
+            cohort_b = Cohort.objects.create(
+                team=self.team,
+                name="Cohort B",
+                groups=[{"properties": [{"key": "id", "value": cohort_a.id, "type": "cohort"}]}],
+                is_static=False,
+            )
+
+            # Create cohort C that depends on cohort B
+            cohort_c = Cohort.objects.create(
+                team=self.team,
+                name="Cohort C",
+                groups=[{"properties": [{"key": "id", "value": cohort_b.id, "type": "cohort"}]}],
+                is_static=False,
+            )
+
```

Comment:
Would it make sense to make a more complicated chain for testing. For example, perhaps `cohort_c` could depend on `cohort_b` AND `cohort_a` and `cohort_d` only depends on `cohort_b` and make sure it all gets executed correctly.

It's not a show stopper, but just an idea for more robust testing.